src/lib/utils/resolvers/resolveId.ts

```diff
@@ -0,0 +1,13 @@
+import { SnowflakeRegex } from '@sapphire/discord-utilities';
+import type { CommandInteraction, CommandInteractionResolvedData, Snowflake } from 'discord.js';
+
+export type IdResolvable = string | CommandInteraction;
+
+export function resolveId(parameter: IdResolvable, mentionRegex: RegExp, key: keyof CommandInteractionResolvedData): Snowflake | null {
+	if (typeof parameter == 'string') return mentionRegex.exec(parameter)?.[1] ?? SnowflakeRegex.exec(parameter)?.[1] ?? null;
+	if (!parameter.inCachedGuild()) {
+		// TODO: figure out how to get uncached members
```

Comment:
I need some help with this. I tried looking into this but I don't know how to get a guild member from a `APIInteractionDataResolvedGuildMember`. It doesn't have an ID, which I can't tell if Discord didn't think their API through or there's something wrong with the type. Regardless, I would appreciate any help with this.

Reply:
If you get that type, that means there's a user too! Also, it should be keyed by the user id in the raw options 👀